content/news/buzz/2025-curtin-capstone-year-end-blog/contents.lr

```diff
@@ -0,0 +1,104 @@
+
+### Briefcase Web Development Optimisations
+
+We also laid the groundwork for web platform development mode support in Briefcase by making the `dev` command platform-aware, and implementing virtual environment management. We [refactored the core `DevCommand` architecture](https://github.com/beeware/briefcase/pull/2419) by introducing platform-specific `DevCommand` subclasses like `StaticWebDevCommand` to handle platform-dependent workflows). Next, we [implemented virtual environment management](https://github.com/beeware/briefcase/pull/2420) by introducing the VirtualEnvironment context manager tool, which provides isolated, configurable virtual environments for development mode, and [improved the tracking of virtual environment usage by the dev command](https://github.com/beeware/briefcase/pull/2495). We also [modified the handling of local Python assets](https://github.com/beeware/briefcase/pull/2498) so they are installed in editable mode. This work establishes the foundation for exposing the app assets to a development web server allowing the live-reload workflow, ultimately speeding up the web development process. 
+
+### PyScript Briefcase and Toga Dependencies
+
+Lastly, we worked on redesigning the Briefcase static web build pipeline to implement a deterministic asset insertion system which replaced template-based asset management. The new system enables Toga and other toolkits to embed their configuration files and front-end resources directly [into wheel packages](https://github.com/beeware/toga/pull/3666) which the [new Briefcase functionality](https://github.com/beeware/briefcase/pull/2442) will extract and insert into static web files. The [web template received a redesign](https://github.com/beeware/briefcase-web-static-template/pull/21) which established specific areas for insertion. The [previous restrictions](https://github.com/beeware/briefcase/issues/2337) prevented Toga and other toolkits from taking full control of their front-end operations while working independently from Briefcase. The new system separates tasks while making maintenance easier and enables toolkits to manage their front-end components, with Toga now owning its own PyScript version, PyScript configuration and Shoelace HTML script.
+
+## Lessons Learned
+
+This year with BeeWare has marked many valuable lessons taught outside of the class environment. Each of us have now made significant contributions to an open-source project while meeting deadlines and attending weekly standup meetings. This process has allowed us to develop our skills in numerous ways. Below each of us will comment our individual lessons.
```

Comment:
Open Source is a proper noun, because it's a trademark of a very specific set of software licensing criteria:
```suggestion
This year with BeeWare has marked many valuable lessons taught outside of the class environment. Each of us have now made significant contributions to an Open Source project while meeting deadlines and attending weekly standup meetings. This process has allowed us to develop our skills in numerous ways. Below each of us will comment our individual lessons.
```